common/kernel/timing.cc

```diff
@@ -69,6 +58,25 @@ void TimingAnalyser::run(bool update_route_delays)
     walk_backward();
     compute_slack();
     compute_criticality();
+
+    // Ensure we clear all timing results if any of them has been marked as
+    // as to be updated. This is done so we ensure it's not possible to have
+    // timing_result which contains mixed reports
+    if (update_net_timings || update_histogram || update_crit_paths) {
+        result = TimingResult();
+    }
+
+    if (update_net_timings) {
+        build_detailed_net_timing_report();
```

Comment:
@gatecat I have split every single report into it's own step. This means it needs to walk through the graph a few more times. I think it's worth it to do that to make things more clear. But if you want me to merge these together into `walk_forward/walk_backward` that's possible as well.

Reply:
That seems fine to me - the report printing is done rarely enough that this is fine